tests/Fixtures/Maker/expected/can_create_factory_with_embeddable_with_data_set_odm.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Factory;
+
+use Zenstruck\Foundry\ModelFactory;
+use Zenstruck\Foundry\Proxy;
+use Zenstruck\Foundry\Tests\Fixtures\Document\Post;
+use Zenstruck\Foundry\Tests\Fixtures\Factories\ODM\CommentFactory;
+use Zenstruck\Foundry\Tests\Fixtures\Factories\ODM\UserFactory;
+
+/**
+ * @extends ModelFactory<Post>
+ *
+ * @method        Post|Proxy create(array|callable $attributes = [])
+ * @method static Post|Proxy createOne(array $attributes = [])
+ * @method static Post|Proxy find(object|array|mixed $criteria)
+ * @method static Post|Proxy findOrCreate(array $attributes)
+ * @method static Post|Proxy first(string $sortedField = 'id')
+ * @method static Post|Proxy last(string $sortedField = 'id')
+ * @method static Post|Proxy random(array $attributes = [])
+ * @method static Post|Proxy randomOrCreate(array $attributes = [])
+ * @method static Post[]|Proxy[] all()
+ * @method static Post[]|Proxy[] createMany(int $number, array|callable $attributes = [])
+ * @method static Post[]|Proxy[] createSequence(array|callable $sequence)
+ * @method static Post[]|Proxy[] findBy(array $attributes)
+ * @method static Post[]|Proxy[] randomRange(int $min, int $max, array $attributes = [])
+ * @method static Post[]|Proxy[] randomSet(int $number, array $attributes = [])
+ */
+final class PostFactory extends ModelFactory
+{
+    /**
+     * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#factories-as-services
+     *
+     * @todo inject services if required
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#model-factories
+     *
+     * @todo add your default values here
+     */
+    protected function getDefaults(): array
+    {
+        return [
+            'body' => self::faker()->text(),
+            'comments' => CommentFactory::new()->many(5),
```

Comment:
I'm wondering something: if in `CommandFactory` we have
```
protected function getDefaults(): array
{
    return [
        'post' => PostFactory::new()
    ];
}
```
won't we have an infinite loop? or maybe you've already fixed that case?

Reply:
I don't believe this is accounted for.

Reply:
When we get to auto-creating factories for relationships this will be an issue as well (ie self referencing relationships)

Reply:
follow up [here](https://github.com/zenstruck/foundry/pull/364#pullrequestreview-1193542616) since it is the same problem

Reply:
> I don't believe this is accounted for.

same as in https://github.com/zenstruck/foundry/pull/364#pullrequestreview-1193542616 we already test such case, which works fine. 

in `Factory::create()`, `$this->normalizeAttribute($value, $name)` returns null, but it adds an `afterPersist` callback which forces `Comment::$post` to be the current `$post`. This way, there is no infinite loop, and that's how the current object is set to the object owner of the relation.

It took me some time to understand _how_ this works :sweat_smile: - but now I understand things better :blush:

Reply:
About self-referencing relationships, it isn't possible to have a non-nullable self-referencing relationship, is it?

Reply:
hmmm :thinking: 

nothing stops us to do it, even if I cannot find any valid usecase
we even have one in our tests : https://github.com/zenstruck/foundry/blob/1.x/tests/Fixtures/Entity/EntityWithRelations.php#L23

Reply:
That example isn't self-referencing though

Reply:
:facepalm: 

you're right, sorry for that.
I've tried a case with self-referencing, non-nullable relationship, and nothing stops from doing this and doctrine validator says everything is OK.

do you want that we make something special for this case?

Reply:
Nah, was just curious.